.circleci/config.yml

```diff
@@ -0,0 +1,182 @@
+# Configuration file for https://circleci.com/gh/angular/material
+
+# Note: YAML anchors allow an object to be re-used, reducing duplication.
+# The ampersand declares an alias for an object, then later the `<<: *name`
+# syntax dereferences it.
+# See http://blog.daemonl.com/2016/02/yaml.html
+# To validate changes, use an online parser, eg. http://yaml-online-parser.appspot.com/
+
+var_1: &docker_image circleci/node:10.15.0-browsers
+var_2: &cache_key angularjs-material-{{ checksum "package-lock.json" }}
+
+# Settings common to each job
+var_3: &job_defaults
+  working_directory: ~/ng
+  docker:
+    - image: *docker_image
+
+# Job step for checking out the source code from GitHub. This also ensures that the source code
+# is rebased on top of master.
+var_4: &checkout_code
+  checkout:
+    # After checkout, rebase on top of master. By default, PRs are not rebased on top of master,
+    # which we want. See https://discuss.circleci.com/t/1662
+    post: git pull --ff-only origin "refs/pull/${CI_PULL_REQUEST//*pull\//}/merge"
+
+# Restores the cache that could be available for the current lock file. The cache
+# includes the node modules.
+var_5: &restore_cache
+  restore_cache:
+    key: *cache_key
+
+# Saves the cache for the current lock file. We store the node modules cache in order to make
+#  subsequent builds faster.
+var_6: &save_cache
+  save_cache:
+    key: *cache_key
+    paths:
+      - "node_modules"
+      - "$HOME/.npm"
+      - "tmp/angular.js"
+
+# Job step that ensures that the node module dependencies are installed and up-to-date. We use
+# NPM with the frozen lockfile option in order to make sure that lock file and package.json are
+# in sync. Unlike in Travis, we don't need to manually purge the node modules if stale because
+# CircleCI automatically discards the cache if the checksum of the lock file has changed.
+var_7: &npm_install
+  run: npm ci
+
+var_8: &rebuild_node_sass
+  run: npm rebuild node-sass
+
+var_9: &install_gulp
+  run: sudo npm i -g gulp@^3
+
+var_10: &store_artifacts
+  store_artifacts:
+    path: dist
+    destination: /angularjs_material/material_build_output
```

Comment:
Do we need this?

Reply:
It's not doing anything useful at this point in time, no.